numpyro/mcmc.py

```diff
@@ -79,12 +89,20 @@ def init_kernel(init_samples,
         hmc_state = HMCState(vv_state.z, vv_state.z_grad, vv_state.potential_energy, 0, 0.,
                              wa_state.step_size, wa_state.inverse_mass_matrix, rng_hmc)
 
+        wa_update = jit(wa_update)
         if run_warmup:
-            hmc_state, _ = jit(fori_loop, static_argnums=(2,))(0, num_warmup_steps, warmup_update,
-                                                               (hmc_state, wa_state))
+            if use_prims:
+                hmc_state, _ = jit(fori_loop, static_argnums=(2,))(0, num_warmup_steps,
+                                                                   warmup_update,
+                                                                   (hmc_state, wa_state))
+            else:
+                n_iter = (num_warmup_steps if isinstance(num_warmup_steps, Iterable)
+                          else range(num_warmup_steps))
+                for i in n_iter:
```

Comment:
Since the only reason why we imagine this will be useful is for diagnostics / progress bar updates, I think we can just use tqdm directly instead of having to change the type of `num_warmup_steps` to support it. That way we can plug in other diagnostics too from `hmc_state`.

Reply:
Do you want to add `tqdm` as a dependent? If that is the case, then I'll add a flag `prog_bar` as in Pyro for it.

Reply:
I think we will provide some default diagnostics at some point, so we might as well add it. It's a light weight dependency!